core/templates/pages/learner-dashboard-page/new-learner-dashboard/community-lessons-tab.component.ts

```diff
@@ -0,0 +1,309 @@
+    completed: false,
+    playlist: false,
+    subscriptions: false
+  };
+  pageNumberInCommunityLessons: number = 1;
+  pageSize: number = 3;
+  startIndexInCommunityLessons: number = 0;
+  endIndexInCommunityLessons: number = 3;
+  pageNumberInPlaylist: number = 1;
+  startIndexInPlaylist: number = 0;
+  endIndexInPlaylist: number = 3;
+
+  ngOnInit(): void {
+    this.noCommunityLessonActivity = (
+      (this.incompleteExplorationsList.length === 0) &&
+        (this.completedExplorationsList.length === 0) &&
+        (this.incompleteCollectionsList.length === 0) &&
+        (this.completedCollectionsList.length === 0));
+    this.noPlaylistActivity = (
+      (this.explorationPlaylist.length === 0) &&
+      (this.collectionPlaylist.length === 0));
+    this.totalIncompleteLessonsList.push(
+      ...this.incompleteExplorationsList, ...this.incompleteCollectionsList);
+    this.totalCompletedLessonsList.push(
+      ...this.completedExplorationsList, ...this.completedCollectionsList);
+    this.totalLessonsInPlaylist.push(
+      ...this.explorationPlaylist, ...this.collectionPlaylist);
+    this.allCommunityLessons.push(
+      ...this.incompleteExplorationsList, ...this.incompleteCollectionsList,
+      ...this.completedCollectionsList, ...this.completedExplorationsList);
+    this.displayIncompleteLessonsList = this.totalIncompleteLessonsList.slice(
+      0, 3);
+    this.displayCompletedLessonsList = this.totalCompletedLessonsList.slice(
+      0, 3);
+    if (this.checkMobileView()) {
+      this.displayLessonsInPlaylist = this.totalLessonsInPlaylist;
+    } else {
+      this.displayLessonsInPlaylist = this.totalLessonsInPlaylist.slice(0, 3);
+    }
+    this.displayInCommunityLessons = this.allCommunityLessons;
+    this.selectedSection = 'All';
+    this.dropdownEnabled = false;
+  }
+
+  decodePngURIData(base64ImageData: string): string {
+    return decodeURIComponent(base64ImageData);
+  }
+
+  checkMobileView(): boolean {
+    return this.deviceInfoService.isMobileDevice();
+  }
+
+  enableDropdown(): void {
+    this.dropdownEnabled = !this.dropdownEnabled;
+  }
+
+  changeSection(section: string): void {
+    this.dropdownEnabled = !this.dropdownEnabled;
+    this.selectedSection = section;
+    if (section === 'Completed') {
+      this.displayInCommunityLessons = this.totalCompletedLessonsList;
+    } else if (section === 'Incomplete') {
+      this.displayInCommunityLessons = this.totalIncompleteLessonsList;
+    } else if (section === 'All') {
+      this.displayInCommunityLessons = [];
+      this.displayInCommunityLessons.push(
+        ...this.totalIncompleteLessonsList, ...this.totalCompletedLessonsList);
+    }
+    this.pageNumberInCommunityLessons = 1;
+    this.startIndexInCommunityLessons = 0;
+    this.endIndexInCommunityLessons = 3;
+  }
+
+  getLessonType(tile: LearnerExplorationSummary | CollectionSummary): string {
+    if (this.totalIncompleteLessonsList.includes(tile)) {
+      return 'Incomplete';
```

Comment:
Can all these constant strings (here and in below functions) be put in local constants so one of them isn't changed by accident?

Reply:
Done.